llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp

```diff
@@ -876,6 +874,32 @@ struct RISCVOperand final : public MCParsedAsmOperand {
     return isUImmPred([](int64_t Imm) { return isUInt<16>(Imm) && Imm != 0; });
   }
 
+  bool isSImm20LI() const {
+    if (!isImm())
+      return false;
+
+    int64_t Imm;
+    if (evaluateConstantImm(getImm(), Imm))
+      return isInt<20>(fixImmediateForRV32(Imm, isRV64Imm()));
+
+    RISCVMCExpr::Specifier VK = RISCVMCExpr::VK_None;
+    return RISCVAsmParser::classifySymbolRef(getImm(), VK) &&
+           VK == RISCVMCExpr::VK_QC_ABS20;
+  }
+
+  bool isSImm32() const {
```

Comment:
```suggestion
  bool isBareSImm32() const {
```

This would match up with how upstream now names these methods.